src/libsyntax/ext/deriving/rand.rs

```diff
@@ -24,6 +24,9 @@ pub fn expand_deriving_rand<F>(cx: &mut ExtCtxt,
                                push: F) where
     F: FnOnce(P<Item>),
 {
+    cx.span_warn(span,
+                 "`derive(Rand)` is deprecated in favour of `derive_rand` on crates.io");
```

Comment:
Has this crate not been pushed to `rust-lang/rand` just yet? I see it on crates.io but not in git